pkg/operator/staticpod/controller/missingstaticpod/missing_static_pod_controller.go

```diff
@@ -0,0 +1,274 @@
+		}
+	}
+
+	if len(errors) > 0 {
+		return fmt.Errorf(strings.Join(errors, "\n"))
+	}
+
+	return nil
+}
+
+// getStaticPodTerminationGracePeriodSecondsForRevision reads the static pod manifest from a configmap
+// in the target namespace and returns the value of terminationGracePeriodSeconds.
+// In case no value was provided for the static pod it returns a default value of 30 seconds.
+func (c *missingStaticPodController) getStaticPodTerminationGracePeriodSecondsForRevision(revision int) (time.Duration, error) {
+	staticPodKeyName := "pod.yaml"
+	staticPodConfigMapName := fmt.Sprintf("%s-pod-%d", c.operandName, revision)
+
+	staticPodConfigMap, err := c.configMapListerForTargetNamespace.Get(staticPodConfigMapName)
+	if err != nil {
+		return 0, err
+	}
+
+	rawStaticPodManifest, hasPodKey := staticPodConfigMap.Data[staticPodKeyName]
+	if !hasPodKey {
+		return 0, fmt.Errorf("didn't find required key %q in cm: %s/%s", staticPodKeyName, staticPodConfigMapName, c.targetNamespace)
+	}
+
+	serializedStaticPod, err := resourceread.ReadPodV1([]byte(rawStaticPodManifest))
+	if err != nil {
+		return 0, err
+	}
+
+	if serializedStaticPod.Spec.TerminationGracePeriodSeconds == nil {
+		klog.V(6).Infof("optional field: %s.spec.terminationGracePeriodSeconds was not specified in cm: %s, returning default value of 30s", staticPodKeyName, staticPodConfigMapName)
+		return 30 * time.Second, nil
+	}
+
+	return time.Duration(*serializedStaticPod.Spec.TerminationGracePeriodSeconds) * time.Second, nil
+}
+
+func getStaticPodCurrentRevision(node string, status *operatorv1.StaticPodOperatorStatus) int {
+	var nodeStatus *operatorv1.NodeStatus
+	for i := range status.NodeStatuses {
+		if status.NodeStatuses[i].NodeName == node {
+			nodeStatus = &status.NodeStatuses[i]
+			break
+		}
+	}
+
+	if nodeStatus == nil {
+		return 0
+	}
+	return int(nodeStatus.CurrentRevision)
+}
+
+func getMostRecentInstallerPodByNode(pods []*corev1.Pod) (map[string]*corev1.Pod, error) {
+	mostRecentInstallerPodByNode := map[string]*corev1.Pod{}
+	byNodes, err := getInstallerPodsByNode(pods)
+	if err != nil {
+		return nil, err
+	}
+
+	for node, installerPodsOnThisNode := range byNodes {
+		if len(installerPodsOnThisNode) == 0 {
+			continue
+		}
+		sort.Sort(byRevision(installerPodsOnThisNode))
+		mostRecentInstallerPodByNode[node] = installerPodsOnThisNode[len(installerPodsOnThisNode)-1]
+	}
+
+	return mostRecentInstallerPodByNode, nil
+}
+
+func getInstallerPodsByNode(pods []*corev1.Pod) (map[string][]*corev1.Pod, error) {
+	byNodes := map[string][]*corev1.Pod{}
+	for i := range pods {
+		pod := pods[i]
+		if !strings.HasPrefix(pod.Name, "installer-") {
+			continue
+		}
+
+		nodeName := pod.Spec.NodeName
+		if len(nodeName) == 0 {
+			return nil, fmt.Errorf("node name for installer pod %q is empty", pod.Name)
+		}
+		byNodes[nodeName] = append(byNodes[nodeName], pod)
+	}
+
+	return byNodes, nil
+}
+
+type byRevision []*corev1.Pod
```

Comment:
We're copying code from https://github.com/openshift/library-go/blob/18f172cff934d86b5fe70f0f26d101c97a6121e9/pkg/operator/staticpod/installerpod/cmd.go#L553-L583 why not creating a package with just that?

Reply:
good point, I opened https://github.com/openshift/library-go/pull/1317